bot/src/discord/commands/DynamicSoundCommand.ts

```diff
@@ -0,0 +1,166 @@
+import {
+  ActionRowBuilder,
+  ApplicationCommandOptionType,
+  ButtonBuilder,
+  ButtonStyle,
+  ChatInputCommandInteraction,
+  MessageActionRowComponentBuilder,
+  PermissionFlagsBits,
+} from 'discord.js';
+import { prismaClient } from '../../bot';
+import { DEFAULT_AIRHORN_MAX_QUEUE_ITEMS, enqueSound, getTotalItemsInGuildQueue } from '../../utils/AirhornAudio';
+import { DiscordChatInputCommand } from '../types/DiscordChatInputCommand';
+
+export class DynamicSoundCommand extends DiscordChatInputCommand {
+  constructor() {
+    super({
+      name: 'dynamicsound',
+      description: 'see src/utils/RegisterCommandsUtils.ts',
+      options: [
+        {
+          name: 'variant',
+          description: 'Choose the sound variant.',
+          type: ApplicationCommandOptionType.Integer,
+        },
+      ],
+    });
+  }
+
+  async handle(commandInteraction: ChatInputCommandInteraction): Promise<unknown> {
+    if (!commandInteraction.member || !commandInteraction.guildId) {
+      return commandInteraction.reply({
+        content: 'You cannot trigger the bot in a direct message.',
+        ephemeral: true,
+      });
+    }
+    if (!commandInteraction.guild) {
+      return commandInteraction.reply({
+        content: 'The bot must be in the server, try to re-invite it.',
+        ephemeral: true,
+      });
+    }
+    // Get the bot member in the guild
+    const botGuildMember = commandInteraction.guild.members.me;
+    if (!botGuildMember) {
+      return commandInteraction.reply({
+        content: 'The bot was not found in the server.',
+        ephemeral: true,
+      });
+    }
+    // Find the sound command
+    const soundCommand = await prismaClient.soundCommand.findFirst({
+      where: {
+        name: commandInteraction.commandName,
+      },
+    });
+    if (!soundCommand) {
+      return commandInteraction.reply({
+        content: 'No sound was found for this command.',
+        ephemeral: true,
+      });
+    }
+    if (soundCommand.disabled) {
+      return commandInteraction.reply({
+        content: 'The sound command requested is disabled.',
+        ephemeral: true,
+      });
+    }
+    // Find the sound variants for the command
+    const soundsForSoundCommand = await prismaClient.sound.findMany({
+      where: {
+        soundCommandId: soundCommand.id,
+        disabled: false,
+      },
+    });
+    if (soundsForSoundCommand.length === 0) {
+      return commandInteraction.reply({
+        content: 'No sounds were found for the command requested.',
+        ephemeral: true,
+      });
+    }
+    let selectedVariant = soundsForSoundCommand[Math.floor(Math.random() * soundsForSoundCommand.length)];
+    // Check to see if a sound variant is specified (if it is, set the selected variant to the correct one)
+    const variantOption = commandInteraction.options.getInteger('variant', false);
+    if (variantOption !== null) {
```

Comment:
Being nit picky, but do we specifically care if it's null, or should we check undefined too?

Reply:
Not really, changed.